pkg/backend/display/object_diff_test.go

```diff
@@ -134,7 +134,97 @@ func Test_PrintObject(t *testing.T) {
 			t.Parallel()
 
 			var buf bytes.Buffer
-			PrintObject(&buf, c.object, false, 0, deploy.OpSame, false, false, false)
+			PrintObject(&buf, c.object, false, 0, deploy.OpSame, false, false, false, false)
+			assert.Equal(t, c.expected, buf.String())
+		})
+	}
+}
+
+func Test_PrintObjectNoShowSecret(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name     string
+		object   resource.PropertyMap
+		expected string
+	}{
+		{
+			"numbers",
+			resource.NewPropertyMapFromMap(map[string]interface{}{
+				"int":         1,
+				"float":       2.3,
+				"large_int":   1234567,
+				"large_float": 1234567.1234567,
+				"secret":      resource.NewSecretProperty(&resource.Secret{Element: resource.NewStringProperty("secrets")}),
+				"nested_secret": resource.NewPropertyMapFromMap(map[string]interface{}{
+					"super_secret": resource.NewSecretProperty(&resource.Secret{Element: resource.NewStringProperty("super_secret")}),
+				}),
+			}),
+			`<{%reset%}>float        : <{%reset%}><{%reset%}>2.3<{%reset%}><{%reset%}>
+<{%reset%}><{%reset%}>int          : <{%reset%}><{%reset%}>1<{%reset%}><{%reset%}>
+<{%reset%}><{%reset%}>large_float  : <{%reset%}><{%reset%}>1.2345671234567e+06<{%reset%}><{%reset%}>
+<{%reset%}><{%reset%}>large_int    : <{%reset%}><{%reset%}>1234567<{%reset%}><{%reset%}>
+<{%reset%}><{%reset%}>nested_secret: <{%reset%}><{%reset%}>{
+<{%reset%}><{%reset%}>    super_secret: <{%reset%}><{%reset%}>[secret]<{%reset%}><{%reset%}>
+<{%reset%}><{%reset%}>}<{%reset%}><{%reset%}>
+<{%reset%}><{%reset%}>secret       : <{%reset%}><{%reset%}>[secret]<{%reset%}><{%reset%}>
+<{%reset%}>`,
+		},
+	}
+
+	for _, c := range cases {
+		c := c
+		t.Run(c.name, func(t *testing.T) {
+			t.Parallel()
+
+			var buf bytes.Buffer
+			PrintObject(&buf, c.object, false, 0, deploy.OpSame, false, false, false, false)
+			assert.Equal(t, c.expected, buf.String())
+		})
+	}
+}
+
+func Test_PrintObjectShowSecret(t *testing.T) {
```

Comment:
Given this is the same test data I'd probably collapse these into one function and just use `t.Run` to run two subtests, once with show=true, then with false.